crates/bevy_pbr/src/lib.rs

```diff
@@ -211,117 +181,28 @@ impl Default for PbrPlugin {
-            app,
-            SHADOW_SAMPLING_HANDLE,
-            "render/shadow_sampling.wgsl",
-            Shader::from_wgsl
-        );
-        load_internal_asset!(
-            app,
-            PBR_FUNCTIONS_HANDLE,
-            "render/pbr_functions.wgsl",
-            Shader::from_wgsl
-        );
-        load_internal_asset!(
-            app,
-            RGB9E5_FUNCTIONS_HANDLE,
-            "render/rgb9e5.wgsl",
-            Shader::from_wgsl
-        );
-        load_internal_asset!(
-            app,
-            PBR_AMBIENT_HANDLE,
-            "render/pbr_ambient.wgsl",
-            Shader::from_wgsl
-        );
-        load_internal_asset!(
-            app,
-            PBR_FRAGMENT_HANDLE,
-            "render/pbr_fragment.wgsl",
-            Shader::from_wgsl
-        );
-        load_internal_asset!(app, PBR_SHADER_HANDLE, "render/pbr.wgsl", Shader::from_wgsl);
-        load_internal_asset!(
-            app,
-            PBR_PREPASS_FUNCTIONS_SHADER_HANDLE,
-            "render/pbr_prepass_functions.wgsl",
-            Shader::from_wgsl
-        );
-        load_internal_asset!(
-            app,
-            PBR_PREPASS_SHADER_HANDLE,
-            "render/pbr_prepass.wgsl",
-            Shader::from_wgsl
-        );
-        load_internal_asset!(
-            app,
-            PARALLAX_MAPPING_SHADER_HANDLE,
-            "render/parallax_mapping.wgsl",
-            Shader::from_wgsl
-        );
-        load_internal_asset!(
-            app,
-            VIEW_TRANSFORMATIONS_SHADER_HANDLE,
-            "render/view_transformations.wgsl",
-            Shader::from_wgsl
-        );
+        load_shader_library!(app, "render/pbr_types.wgsl");
+        load_shader_library!(app, "render/pbr_bindings.wgsl");
+        load_shader_library!(app, "render/utils.wgsl");
+        load_shader_library!(app, "render/clustered_forward.wgsl");
+        load_shader_library!(app, "render/pbr_lighting.wgsl");
+        load_shader_library!(app, "render/pbr_transmission.wgsl");
+        load_shader_library!(app, "render/shadows.wgsl");
+        load_shader_library!(app, "deferred/pbr_deferred_types.wgsl");
+        load_shader_library!(app, "deferred/pbr_deferred_functions.wgsl");
+        load_shader_library!(app, "render/shadow_sampling.wgsl");
+        load_shader_library!(app, "render/pbr_functions.wgsl");
+        load_shader_library!(app, "render/rgb9e5.wgsl");
+        load_shader_library!(app, "render/pbr_ambient.wgsl");
+        load_shader_library!(app, "render/pbr_fragment.wgsl");
+        load_shader_library!(app, "render/pbr.wgsl");
+        load_shader_library!(app, "render/pbr_prepass_functions.wgsl");
+        load_shader_library!(app, "render/pbr_prepass.wgsl");
+        load_shader_library!(app, "render/parallax_mapping.wgsl");
+        load_shader_library!(app, "render/view_transformations.wgsl");
+
         // Setup dummy shaders for when MeshletPlugin is not used to prevent shader import errors.
```

Comment:
This looks wrong. I don't understand why we would need the visibility buffer imported without the MeshletPlugin.

This might actually cause conflicts if this shader is loaded at the same time as the Meshlet stuff is (because you'll have two shaders with the same import path I'm guessing).

Reply:
thats how it was before, its just an empty shader. its needed because naga_oil isnt very discerning, and validates import paths even in ifdef'd out code. it'll get better with wesl probably

Reply:
This dummy thing works because it first inserts this dummy one, then later the MeshletPlugin inserts the real one to replace it **while building the App**. So the shader cache only sees one asset with that import path - the real one.

Since we're loading with embedded assets and internal assets, there are actually two shaders with the same import path. I'm not sure what the shader cache will do, but at best it will do what we want by chance (which means it's very brittle).

The current state is already brittle. I bet if we added the MeshletPlugin before this plugin, it would also break.

I'm not sure if you have another idea here, but I think we should continue to use internal assets for this one line. I think we're keeping meshlets the same anyway in this PR.

Reply:
makes sense! i'll fix that when i have time

Reply:
done